network/iptables.go

```diff
@@ -202,19 +254,40 @@ func DeleteIP6Tables(rules []IPTablesRule) error {
 	return nil
 }
 
-func ensureIPTables(ipt IPTables, rules []IPTablesRule) error {
+func ensureIPTables(ipt IPTables, rules []IPTablesRule, rulesToDelete []IPTablesRule) error {
 	exists, err := ipTablesRulesExist(ipt, rules)
 	if err != nil {
 		return fmt.Errorf("Error checking rule existence: %v", err)
 	}
-	if exists {
+	ensureDelete := make([]IPTablesRule, 0)
+	for _, ruleToDelete := range rulesToDelete {
+		ruleExist := false
+		for _, rule := range rules {
+			if rule.Equal(&ruleToDelete) {
+				ruleExist = true
+				break
+			}
+		}
```

Comment:
What happens if we try to delete them and they don't exist? I wonder if that might be better than checking if they exist or not

Reply:
OK, Added rule check before add to rulelist.